src/Resources/public/contao-swiper.js

```diff
@@ -0,0 +1,51 @@
+(function() {
+    // check if a sliderConfig was created inside the templates
+    if (typeof sliderConfig !== 'undefined') {
+        var sliders = Object.keys(sliderConfig);
+        sliders.forEach(function (sliderId) {
+            // get properties for this slider
+            var cssClass = sliderConfig[sliderId]['cssclass'];
+            var params = sliderConfig[sliderId]['params'];
+
+            // adjust html structure
+            var slider = document.getElementById(sliderId);
+            var sliderElement = slider.querySelector('.swiper-container');
+            sliderElement.innerHTML = '<div class="swiper-wrapper">' + sliderElement.innerHTML + '</div>'; // add the swiper-wrapper element inside the swiper-container
+            var wrapperElement = sliderElement.querySelector('.swiper-wrapper');
+            if (cssClass) {
+                // add custom css class
+                wrapperElement.classList.add(cssClass);
+            }
```

Comment:
Now I get the intention of this.
Now, when there is a value for the custom-wrapper class, I check if the next element inside the slider-container has this class.
This element will then be the slider-wrapper.

Reply:
It does not necessarily have to be the next element. It can be nested deeper too. My original jQuery code was this:
```php
<?php if (!$this->wrapperClass): ?>
  $('<?= $this->container ?> > *').wrapAll($('<div class="swiper-wrapper"></div>'));
<?php else: ?>
  $('<?= $this->container ?> .<?= $this->wrapperClass ?>').addClass('swiper-wrapper');
<?php endif; ?>
```
The wrapper class can be _anywhere within the container_. This needs to be kept the same, otherwise it breaks the backwards compatibility.

Reply:
So now the element with the given class is searched inside the container. The first occurrence will be used. It does not need to be the very next element.

Reply:
👍 looks good :)